lib/pbench/server/api/resources/query_apis/iteration_samples.py

```diff
@@ -122,13 +124,22 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
         # table
         try:
             index_map = Metadata.getvalue(dataset=dataset, key="server.index-map")
-            for key, values in index_map.items():
-                if run_id in values:
-                    index = key
-                    self.logger.debug(f"Iteration samples index, {index!r}")
+            index_keys = [key for key in index_map if "result-data-sample" in key]
         except MetadataError as e:
             abort(HTTPStatus.BAD_REQUEST, message=str(e))
 
+        if not len(index_keys) == 1:
+            self.logger.error(
+                f"Found irregular result-data-sample indices for a dataset "
+                f"{dataset.controller!r}|{dataset.name!r}"
+            )
```

Comment:
If you saw this error in the log, wouldn't you want to know what the irregular values _were_??

I think you should include the list in the message.